src/timesync/timesyncd-bus.c

```diff
@@ -43,6 +45,58 @@ static int property_get_servers(
         return sd_bus_message_close_container(reply);
 }
 
+static int method_set_runtime_servers(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_strv_free_ char **valid_names = NULL;
+        _cleanup_strv_free_ char **msg_names = NULL;
+        Manager *m = userdata;
+        int r;
+
+        assert(m);
+        assert(message);
+
+        r = sd_bus_message_read_strv(message, &msg_names);
+        if (r < 0)
+                return r;
+
+        STRV_FOREACH(name, msg_names) {
+                r = dns_name_is_valid_or_address(*name);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to check validity of NTP server name or address '%s': %m", *name);
+
+                if (r == 0) {
+                        log_error("Invalid NTP server name or address, ignoring: %s", *name);
+                        continue;
+                }
+
+                r = strv_extend(&valid_names, *name);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add runtime server '%s': %m", *name);
+        }
+
+        manager_flush_runtime_servers(m);
+
+        STRV_FOREACH(name, valid_names) {
+                r = server_name_new(m, NULL, SERVER_RUNTIME, *name);
+                if (r < 0) {
+                        manager_flush_runtime_servers(m);
+
+                        return log_error_errno(r, "Failed to add runtime server '%s': %m", *name);
+                }
+        }
+
+        m->runtime_servers_changed = true;
+
+        if (manager_is_connected(m)) {
+                if (m->current_server_address &&
+                    m->current_server_address->name &&
+                    strv_contains(valid_names, m->current_server_address->name->string))
+                        (void) manager_connect(m);
+        } else
+                (void) manager_connect(m);
```

Comment:
seems you can merge the inner and the outer if check? they execute the same operation in the inner and the else branch after all?